test/e2e/tests/add-custom-network.spec.js

```diff
@@ -17,6 +17,232 @@ describe('Custom network', function () {
       },
     ],
   };
+
+  it('should show warning when adding chainId 0x1', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        await driver.executeScript(`
+          var params = [{
+            chainId: "0x1",
+            chainName: "Fake Ethereum Network",
+            nativeCurrency: {
+              name: "",
+              symbol: "ETH",
+              decimals: 18
+            },
+            rpcUrls: ["https://customnetwork.com/api/customRPC"],
+            blockExplorerUrls: [ "http://localhost:8080/api/customRPC" ]
+          }]
+          window.ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params
+          })
+        `);
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          3,
+          1000,
+          10000,
+        );
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Approve',
+        });
+
+        await driver.delay(1000);
+        const warningTxt =
+          'You are adding a new RPC provider for Ethereum Mainnet';
+
+        const warning = await driver.findElement({
+          tag: 'h4',
+          text: warningTxt,
+        });
+
+        assert.equal(await warning.getText(), warningTxt);
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Approve',
+        });
+
+        const errMsg =
+          'Chain ID returned by the custom network does not match the submitted chain ID.';
+        const error = await driver.findElement({
+          tag: 'span',
+          text: errMsg,
+        });
+        assert.equal(await error.getText(), errMsg);
```

Comment:
Nit: These assertions are redundant; the `findElement` call will throw unless it finds something with that exact text. It's already asserting this for us. Removing these assertions saves another round-trip communication with the browser, reducing test runtime and flakiness.

```suggestion
        await driver.findElement({
          tag: 'h4',
          text: warningTxt,
        });

        await driver.clickElement({
          tag: 'button',
          text: 'Approve',
        });

        const errMsg =
          'Chain ID returned by the custom network does not match the submitted chain ID.';
        await driver.findElement({
          tag: 'span',
          text: errMsg,
        });
```

Reply:
I see there are a few examples of this in the other tests as well